orchagent/fdborch.h

```diff
@@ -5,29 +5,70 @@
 #include "observer.h"
 #include "portsorch.h"
 
+enum FdbOrigin
+{
+    FDB_ORIGIN_INVALID = 0,
+    FDB_ORIGIN_LEARN = 1,
+    FDB_ORIGIN_PROVISIONED = 2,
+    FDB_ORIGIN_VXLAN_ADVERTIZED = 4
+};
+
 struct FdbEntry
 {
     MacAddress mac;
     sai_object_id_t bv_id;
-    std::string port_name;
 
     bool operator<(const FdbEntry& other) const
     {
         return tie(mac, bv_id) < tie(other.mac, other.bv_id);
     }
+    bool operator==(const FdbEntry& other) const
+    {
+        return tie(mac, bv_id) == tie(other.mac, other.bv_id);
+    }
 };
 
 struct FdbUpdate
 {
     FdbEntry entry;
     Port port;
+    string type;
     bool add;
 };
 
+struct FdbData
+{
+    sai_object_id_t bridge_port_id;
+    string type;
+    FdbOrigin origin;
+    /**
+      {"dynamic", FDB_ORIGIN_LEARN} => dynamically learnt
+      {"dynamic", FDB_ORIGIN_PROVISIONED} => provisioned dynamic with swssconfig in APPDB
+      {"dynamic", FDB_ORIGIN_ADVERTIZED} => synced from remote device e.g. BGP MAC route
+      {"static", FDB_ORIGIN_LEARN} => Invalid
+      {"static", FDB_ORIGIN_PROVISIONED} => statically provisioned
+      {"static", FDB_ORIGIN_ADVERTIZED} => sticky synced from remote device
+    */
+
+    /* Remote FDB related info */
+    string remote_ip;
+    string    esi;
+    unsigned int vni;
+};
+
 struct SavedFdbEntry
 {
-    FdbEntry entry;
+    MacAddress mac;
+    unsigned short vlanId;
     string type;
```

Comment:
Can we remove all the variable below this variable and add fdbData as a variable?

Reply:
Done